docs/configuration.rst

```diff
@@ -2650,3 +2438,312 @@ final need and schema looks like.
+             "field3": {
+                 "predicates": [('status == "done"', "a")],
+                 "default": "b",
+             },
+             "field4": {
+                 "predicates": [
+                     ('status == "done"', "a"),
+                     ('status == "ongoing"', "b"),
+                     ('status == "other"', "c"),
+                 ],
+                 "default": "d",
+             },
+         }
+
+This configuration allows for global defaults to be set for all needs,
+for any of the following fields:
+
+- any :ref:`needs_fields` key
+- any ``needs_extra_links`` field
+- ``status``
+- ``layout``
+- ``style``
+- ``tags``
+- ``constraints``
+
+Defaults will be used if the field is not set specifically by the user and thus has a "empty" value.
+
+.. code-block:: python
+
+   needs_fields = {"option1": {}}
+   needs_global_options = {
+      "tags": {"default": ["tag1", "tag2"]},
+      "option1": {"default": "new value"},
+   }
+
+To set a default based on a one or more predicates, use the ``predicates`` key.
+These predicates are a list of ``(match expression, value)``, evaluated in order, with the first match set as the default value.
+
+A match expression is a string, using Python syntax, that will be evaluated against data from each need (before the resolution of defaults or dynamic functions etc):
+
+- ``id`` (``str``)
+- ``type`` (``str``)
+- ``title`` (``str``)
+- ``tags`` (``tuple[str, ...]``)
+- ``status`` (``str | None``)
+- ``docname`` (``str | None``)
+- ``is_external`` (``bool``)
+- ``is_import`` (``bool``)
+- :ref:`needs_fields`
+- :ref:`needs_extra_links` (``tuple[str, ...]``)
+- :ref:`needs_filter_data`
+
+If no predicates match, the ``default`` value is used (if present).
+
+.. code-block:: python
+
+   needs_fields = {"option1": {}}
+   needs_global_options = {
+      "option1": {
+      # if field is unset:
+         "predicates": [
+            # if status is "done", set to "value1"
+            ("status == 'done'", "value1"),
+            # else if status is "ongoing", set to "value2"
+            ("status == 'ongoing'", "value2"),
+         ]
+         # else, set to "value3"
+         "default": "value3",
+      }
+   }
+
+.. tip::
+
+   You can combine global options with :ref:`dynamic_functions` to automate data handling.
+
+   .. code-block:: python
+
+      needs_fields = {"option1": {}}
+      needs_global_options = {
+              "option1": {"default": '[[copy("id")]]'}
+      }
+
+.. _`needs_statuses`:
+
+needs_statuses
+~~~~~~~~~~~~~~
+
+.. deprecated:: 7.0.0
+
+   Use :ref:`needs_fields` instead.
+
+Defines a set of valid statuses, which are allowed to be used inside documentation.
+If we detect a status not defined, an error is thrown and the build stops.
+The checks are case sensitive.
+
+Activate it by setting it like this:
+
+.. code-block:: python
+
+   needs_statuses = [
+       dict(name="open", description="Nothing done yet"),
```

Comment:
We are losing out on the description after the deprecation. Was this used for anything?
That could also be part of the schema description, if that's the case.

Reply:
No it wasn't used for anything in particular.
It's always nice to be able to document enum variants (as would be the rust terminology) but how can you do this in the schema? You could use oneOf + const, but that gets messy to "analyse"

```json

{
  "type": "string",
  "oneOf": [
    {
      "const": "draft",
      "description": "Work in progress; not yet reviewed"
    },
    {
      "const": "approved",
      "description": "Reviewed and approved for use"
    },
    {
      "const": "rejected",
      "description": "Reviewed and explicitly rejected"
    }
  ]
}

```